### ISSUE TITLE
Use pyproject.toml

### ISSUE THREAD

**Issue by peterdudfield** [openclimatefix/neso-solar-consumer#9]
Good to switch to pyproject.toml instead of setup.py

**Comment by alirashidAR** [comment]
I’m interested in working on this issue. Could you please assign it to me or let me know if there are any specific guidelines or prerequisites? 

Thank you !

**Comment by siddharth7113** [comment]
Hi @peterdudfield and @alirashidAR,

I’d like to apologize for not requesting assignment before starting work on this issue—this was my oversight. While reviewing the issue, I went ahead and completed the migration to `pyproject.toml` and updated some of the project structure file. I now see the issue has been assigned to @alirashidAR.

Since I have already completed the work, would it be okay for me to submit a pull request for review? I’d be happy to collaborate with @alirashidAR  if needed or incorporate any additional changes they’d like to make.

Please let me know how you’d like to proceed, and I’ll follow your guidance. Thank you!

**Comment by alirashidAR** [comment]
Hi @siddharth7113 ,

I have made the subsequent changes , I'll create a pull request and if you have any suggestions then we can collaborate on it . I did not change the file structure , simply followed a flat-layout .

**Comment by peterdudfield** [comment]
Thanks @siddharth7113 and @alirashidAR for both working on this. Sorry there is always a slight danger of this situation happening, but normally tehre are code improvements from both contributors. 

@siddharth7113 do you want to review #12 and you can add your added changed on top in a new PR?

**Comment by siddharth7113** [comment]
Thanks, @peterdudfield! Sure, I'll review #12 and provide feedback. Once that's done, I'll prepare a new PR to add the changes I worked on top of it.
